lms/services/canvas_studio.py

```diff
@@ -0,0 +1,221 @@
+from typing import Literal, Optional, TypedDict
+from urllib.parse import urlencode, urlunparse
+
+from marshmallow import EXCLUDE, Schema, fields, post_load
+
+from lms.models.oauth2_token import Service
+from lms.services.aes import AESService
+from lms.services.oauth_http import factory as oauth_http_factory
+from lms.validation._base import RequestsResponseSchema
+
+
+class CanvasStudioCollectionsSchema(RequestsResponseSchema):
+    """Schema for Canvas Studio /collections responses."""
+
+    class CollectionSchema(Schema):
+        class Meta:
+            unknown = EXCLUDE
+
+        id = fields.Integer(required=True)
+        name = fields.Str(required=True)
+        type = fields.Str(required=True)
+        created_at = fields.Str(required=False)
+
+    collections = fields.List(fields.Nested(CollectionSchema), required=True)
+
+    @post_load
+    def post_load(self, data, **_kwargs):
+        return data["collections"]
+
+
+class CanvasStudioCollectionMediaSchema(RequestsResponseSchema):
+    """Schema for Canvas Studio /collections/{id}/media responses."""
+
+    class MediaSchema(Schema):
+        class Meta:
+            unknown = EXCLUDE
+
+        id = fields.Integer(required=True)
+        title = fields.Str(required=True)
+        created_at = fields.Str(required=False)
+
+    media = fields.List(fields.Nested(MediaSchema), required=True)
+
+    @post_load
+    def post_load(self, data, **_kwargs):
+        return data["media"]
+
+
+class APICallInfo(TypedDict):
+    path: str
+    authUrl: str | None
+
+
+class File(TypedDict):
```

Comment:
I mentioned this in the other PR.

I have a local branch where I expose this outside the LMS-specific services so we can reuses them. I'll try to finish that soon to see how it looks.